### ISSUE TITLE
Private final issue for Ex3

### ISSUE THREAD

**Issue by cher212** [nus-cs2030/2223-s2#213]
I declared the following attributes as protected final so I can access it in the child class, but CodeCrunch needs it to be private final so how should I overcome this issue?
<img width="182" alt="image" src="https://user-images.githubusercontent.com/125173830/220519428-4aa39026-a4ee-46b8-a1ba-a2d36ac9a7d9.png">

**Comment by keshialimm** [comment]
declaring it private final will still allow ur child classes to access it, just use getters?

**Comment by leonardwx** [comment]
yes set it to private final and use getters to abstract the properties

**Comment by cher212** [comment]
Does this also mean I cannot use "super" at all?

**Comment by Klowhe** [comment]
Private methods of the super-class cannot be called. Only public and protected methods can be called by the super keyword.

**Comment by granolaluvr69** [comment]
To add on, you can still use super to reference superclasses' methods, e.g. super.getRadius().

**Comment by e0969533** [comment]
you can call super.method() as long as the method in the parent class is declared as public. as for attributes you can set getter methods in that class.

**Comment by territxl** [comment]
I think creating getter methods would be a way for you to obtain attributes. If you would like to call a method in the parent class, using super.method() would also be possible

**Comment by Cepheux** [comment]
what worked for me was to super the parent class (with all attributes private final) and still declare the attributes for the child class,  (its probably double job though)

**Comment by ryantangmj** [comment]
you could use a getter to access these variables in your child class
